src/js/components/DataFilter/DataFilter.js

```diff
@@ -8,6 +8,10 @@ import { RangeSelector } from '../RangeSelector';
 import { SelectMultiple } from '../SelectMultiple';
 import { DataFilterPropTypes } from './propTypes';
 
+// empirical constants for when we change inputs
+const maxCheckBoxGroupOptions = 7;
+const minSelectSearchOptions = 12;
```

Comment:
When researching SelectMultiple, the team set these thresholds at:
 - CheckBoxGroup = 1-4
 - SelectMultiple = 5-9
 - SelectMultiple with search = 10+
 
Do we want to align with guidance on the HPE Design System ([min](https://design-system.hpe.design/components/selectmultiple#when-not-to-use),[max](https://design-system.hpe.design/components/selectmultiple#best-practices))?
```suggestion
// empirical constants for when we change inputs
const maxCheckBoxGroupOptions = 4;
const minSelectSearchOptions = 10;
```